packages/block-editor/src/components/url-popover/image-url-input-ui.js

```diff
@@ -245,9 +246,11 @@ const ImageURLInputUI = ( {
 				label={ url ? __( 'Edit link' ) : __( 'Insert link' ) }
 				aria-expanded={ isOpen }
 				onClick={ openLinkUI }
+				ref={ buttonRef }
 			/>
 			{ isOpen && (
 				<URLPopover
+					anchorRef={ buttonRef }
```

Comment:
Passing an explicit anchorRef here so that the URL popover is anchored to the button (and not to the whole toolbar).